grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details.ng.html

```diff
@@ -0,0 +1,34 @@
+<div *ngIf="flowRunning$ | async" class="in-progress">
+  {{ (args$ | async).query }}
+</div>
+
+<div *ngIf="flowCompleted$ | async" class='results'>
+  <div *ngIf="resultTable$ | async as resultTable">
+    <div class="results-query">
+      <div class="results-query-label">Query:</div>&nbsp;
+      <div class="results-query-text">{{resultTable.query}}</div>
+    </div>
+
+    <div class="table-container">
+      <table>
+        <thead>
+          <th *ngFor="let column of resultTable.header.columns">{{column.name}}</th>
```

Comment:
add `trackBy: fn` and track the entries by a primitive, not a POJO. Since objects are unequal, unless they are the same reference, Angular would re-render this table even if the data is actually the same.

Reply:
I looked into that, but I couldn't understand exactly under what circumstances our component could get different objects for the same value. Thinking about it now, I guess this could happen if we request data in batches, more than once?

This reminded me I should add support for getting all the data, as currently I'm just getting the first batch.

Also I'm not sure at the moment how often will this function be called. If we have 1m rows, I should probably think about the efficiency of the function? A thought that comes to mind is to use the row content's hash. Chances for collision are small even with 1m rows.